src/daemon/daemon.go

```diff
@@ -867,6 +869,19 @@ func (dm *Daemon) onConnect(e ConnectEvent) {
 	dm.recordIPCount(a)
 
 	if e.Solicited {
+		// Disconnect if the max outgoing connections is reached
+		n, err := dm.Pool.Pool.Size()
+		if err != nil {
+			logger.WithError(err).Error("get connection pool size failed")
+			return
+		}
+
+		if n > dm.Config.OutgoingMax {
```

Comment:
Doesn't pool size include both incoming and outgoing?

Reply:
Oh yes, will update